modules/database/test/std/rec/printfTest.c

```diff
@@ -0,0 +1,573 @@
+    testdbPutFieldOk("test_printf_rec.FMT", DBF_STRING, format_string);
+
+    /* set value on inp0 */
+    testdbPutFieldOk("test_printf_inp0_rec.VAL", DBF_SHORT, 014);
+
+    /* verify that string is formatted as expected */
+    testdbGetFieldEqual("test_printf_rec.VAL", DBF_STRING, result_string);
+
+    // number of tests = 3
+}
+
+static void test_min_width_flag(void){
+
+    const char format_string[] = "Format test string %04i";
+    const char result_string[] = "Format test string 0003";
+
+    /* set format string */
+    testdbPutFieldOk("test_printf_rec.FMT", DBF_STRING, format_string);
+
+    /* set value on inp0 */
+    testdbPutFieldOk("test_printf_inp0_rec.VAL", DBF_SHORT, 3);
+
+    /* verify that string is formatted as expected */
+    testdbGetFieldEqual("test_printf_rec.VAL", DBF_STRING, result_string);
+
+    // number of tests = 3
+}
+
+static void test_prec_flag(void){
+
+    const char format_string[] = "Format test string %.4f";
+    const char result_string[] = "Format test string 71.2000";
+
+    /* set format string */
+    testdbPutFieldOk("test_printf_rec.FMT", DBF_STRING, format_string);
+
+    /* set value on inp0 */
+    testdbPutFieldOk("test_printf_inp0_rec.VAL", DBF_DOUBLE, 71.2);
+
+    /* verify that string is formatted as expected */
+    testdbGetFieldEqual("test_printf_rec.VAL", DBF_STRING, result_string);
+
+    // number of tests = 3
+}
+
+static void test_h_flag(void){
+
+    const char format_string[] = "Format test string %hx";
+    const char result_string[] = "Format test string baba";
+
+    /* set format string */
+    testdbPutFieldOk("test_printf_rec.FMT", DBF_STRING, format_string);
+
+    /* set value on inp0 */
+    testdbPutFieldOk("test_printf_inp0_rec.VAL", DBF_LONG, 0xffbaba);
+
+    /* verify that string is formatted as expected */
+    testdbGetFieldEqual("test_printf_rec.VAL", DBF_STRING, result_string);
+
+    // number of tests = 3
+}
+
+static void test_hh_flag(void){
+
+    const char format_string[] = "Format test string %hhx";
+    const char result_string[] = "Format test string c1";
+
+    /* set format string */
+    testdbPutFieldOk("test_printf_rec.FMT", DBF_STRING, format_string);
+
+    /* set value on inp0 */
+    testdbPutFieldOk("test_printf_inp0_rec.VAL", DBF_LONG, 0xffc0c1);
+
+    /* verify that string is formatted as expected */
+    #ifdef __rtems__
+    testTodoBegin("Fails on UB-20 gcc-9 on RTEMS");
+    #endif
+    testdbGetFieldEqual("test_printf_rec.VAL", DBF_STRING, result_string);
+    #ifdef __rtems__
+    testTodoEnd();
+    #endif
+
+    // number of tests = 3
+}
+
+static void test_l_flag(void){
+
+    const char format_string[] = "Format test string %lx";
+    const char result_string[] = "Format test string a0a1c0c1";
+
+    /* set format string */
+    testdbPutFieldOk("test_printf_rec.FMT", DBF_STRING, format_string);
+
+    /* set value on inp0 */
+    testdbPutFieldOk("test_printf_inp0_rec.VAL", DBF_LONG, 0xba0a1c0c1);
+
```

Comment:
Should that start `0xa0a1` and not `0xba0a1` ?

Reply:
I think that may be the cause of the x86 failure above with `%lx`-  ignore my post about `%llx` as i was looking at the wrong test

Reply:
After above change, i suspect that the following could work to protect just %F test
```
// visual studio less than 2015 does not support %F
// mingw/gcc also fails, suspect this may be gcc version
// related and checking __GNUC__ could resolve but 
// initial attempts didn't work so excluding mingw entirely for now
#ifdef _WIN32
#if (defined(_MSC_VER) && _MSC_VER < 1900) || defined(_MINGW)
testTodoBegin("Fails on windows with old visual studio versions and mingw");
#endif
#endif
```